flux/src/main/java/software/amazon/aws/clients/swf/flux/poller/DecisionTaskPoller.java

```diff
@@ -839,34 +857,61 @@ private static Decision handleWorkflowCompletion(Workflow workflow, String workf
         return decision;
     }
 
+    private static List<Decision> handleUnknownResultCode(WorkflowStep currentStep, String actualResultCode,
+                                                          WorkflowState state, Set<String> validResultCodes) {
+        List<Decision> decisions = new ArrayList<>();
+
+        RecordMarkerDecisionAttributes markerAttrs = RecordMarkerDecisionAttributes.builder()
+                .markerName("UnknownResultCode")
+                .details("Unrecognized result code '" + actualResultCode + "' for workflow step "
+                         + currentStep.getClass().getSimpleName() + ". Valid result codes: "
+                         + String.join(", ", validResultCodes))
+                .build();
+        decisions.add(Decision.builder().decisionType(DecisionType.RECORD_MARKER)
+                              .recordMarkerDecisionAttributes(markerAttrs).build());
+
```

Comment:
it will be useful to emit a metric here so the user can set up alarms based on multiple occurrences of unknown result code that extends their deployment time (since it would be acceptable to have unknown result code during deployments)

Reply:
Oh, good call.